### PR TITLE
Update AzureRM.DataLakeStore.psd1 and AssemblyInfo.cs

### PR DESCRIPTION
**Pull request by rahuldutta90**
## Description
Addressed Cormac's comments in https://github.com/Azure/azure-powershell/pull/4986
 1) Update the ModuleVersion in AzureRM.DataLakeStore.psd1 to 5.0.1 
2) Update the ReleaseNotes field in AzureRM.DataLakeStore.psd1 with the contents that you added to your ChangeLog.md file 
3) Update Prerelease field in AzureRM.DataLakeStore.psd1 to preview
4) Update the versions in AssemblyInfo.cs to 5.0.1 
5) Add Microsoft.Azure.DataLake.Store and NLog to RequiredAssemblies field in the AzureRM.DataLakeStore.psd1 file

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/DataLakeStore/AzureRM.DataLakeStore.psd1 [hunk 1]
```diff
@@ -152,14 +151,45 @@ PrivateData = @{
...
-        ReleaseNotes = '* NOTE: This is a breaking change release. Please see the migration guide (https://aka.ms/azps-migration-guide) for a full list of breaking changes introduced.
-* Removed the Obsolete Properties fields in PSDataLakeStoreAccount.cs and its assoicated files
-* Changed one of the two OutputTypes of Get-AzureRmDataLakeStoreAccount
-    - List\<PSDataLakeStoreAccount> to List\<PSDataLakeStoreAccountBasic>
-    - The properties of PSDataLakeStoreAccountBasic is a strict subset of the properties of PSDataLakeStoreAccount
-    - The additional properties that are in PSDataLakeStoreAccount are not returned by the service.  Therefore, this change is to reflect this accurately. These additional properties are still in PSDataLakeStoreAccountBasic, but they are tagged as Obsolete.
-* Add support for online help
-    - Run Get-Help with the -Online parameter to open the online help in your default Internet browser'
+        ReleaseNotes = '
+		* Moved from AutoRest sdk to our version of Dataplane sdk for Azure data lake.
```
Comment (markcowl) [inline src/ResourceManager/DataLakeStore/AzureRM.DataLakeStore.psd1]: This should be customer-relevant information.  You should specify which cmdlets have changed, and either have a link to the breaking changes document here (if there are many changes), or specify the specific changes (if there are only a few).

Details like the folder structure and the technology used for thedata plane client are irrelevant to customers.

Also, please make sure this information is included in your changelog.

Comment (rahuldutta90) [reply src/ResourceManager/DataLakeStore/AzureRM.DataLakeStore.psd1]: @markcowl Fixed it and put the same contents under ChangeLog.

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/295/

signing run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/66/

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/297/

signing run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/67/

**Pull request merged**